web/src/api/getWeatherData.ts

```diff
@@ -105,25 +105,28 @@ async function getWeatherData(type: WeatherType) {
   return interdata;
 }
 
-const useGetWeather = (
-  type: WeatherType,
-  options?: UseQueryOptions<Maybe<GfsForecastResponse>>
-) => {
-  return useQuery<Maybe<GfsForecastResponse>>(
-    [type],
-    async () => await getWeatherData(type),
-    {
-      staleTime: FIVE_MINUTES,
-      cacheTime: FIVE_MINUTES,
-      retry: false, // Disables retrying as getWeatherData handles retrying with new timestamps
-      ...options,
-    }
-  );
+interface UseWeatherQueryOptions
+  extends Omit<UseQueryOptions<Maybe<GfsForecastResponse>>, 'queryKey'> {
+  type: WeatherType;
+}
+
+const useGetWeather = (options: UseWeatherQueryOptions) => {
+  const { type, ...queryOptions } = options;
+  // Ensure getWeatherData returns Promise<Maybe<GfsForecastResponse>>
+  // and FIVE_MINUTES is a number representing milliseconds
+  return useQuery<Maybe<GfsForecastResponse>>({
+    queryKey: [type],
+    queryFn: () => getWeatherData(type), // Removed unnecessary async/await
```

Comment:
No need to keep this comment
```suggestion
    queryFn: () => getWeatherData(type),
```